zerver/lib/request.py

```diff
@@ -24,6 +27,43 @@
 
 from zerver.lib.exceptions import ErrorCode, InvalidJSONError, JsonableError
 from zerver.lib.types import Validator, ViewFuncT
+from zerver.models import Client, Realm
+
+
+@dataclass
+class ZulipRequestData:
+    """This contains the additional attributes for a request that we
+    add in zerver/lib/middleware.py. Presumably, certain fields are
+    expected to not None in `request_data_map[request]` as long as
+    the request is being accessed after being processed in the middleware.
+
+    """
+
+    client: Optional[Client] = None
+    client_name: Optional[str] = None
+    client_version: Optional[str] = None
+    _log_data: Optional[MutableMapping[str, Any]] = None
+    _rate_limit: Optional[str] = None
+    _requestor_for_logs: Optional[str] = None
+    realm: Optional[Realm] = None
+    _set_language: Optional[str] = None
+    _ratelimits_applied: Optional[List[Any]] = None
```

Comment:
This can be `ratelimits_applied: List[RateLimitResult] = field(default_factory=list)`.

Reply:
Yes, I will push an update for this with a little workaround to avoid cyclic imports brought by `RateLimitResult`, which made me set this as `Any`.